### PR TITLE
Move the description of Iceberg procedures to a dedicated subheading

### PR DESCRIPTION
**Pull request by jackychen718**
According #22608:
Move the description of Iceberg procedures to a dedicated subheading.

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 1]
```diff
@@ -1000,6 +1104,8 @@ Examples:
 
     CALL iceberg.system.expire_snapshots(schema => 'schema_name', table_name => 'table_name', snapshot_ids => ARRAY[10001, 10002]);
 
+=======
+>>>>>>> Move the description of Iceberg procedures to a dedicated subheading
```
Comment (hantangwangd) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:1108]: Here exists a conflict. Please handle it first.

Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst]: conflicts fixed.

**Comment by tdcmeehan** [comment]
@jackychen718 per our [CONTRIBUTING.md](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md), you need to make sure your commits are squashed.  Please follow a resource on the internet if you don't know what this means--[here is one article I found that may help](https://www.freecodecamp.org/news/git-squash-commits/).

I'll mark this as draft, please take it out of draft once you've completed this, thanks!

**Review by steveburnett — CHANGES_REQUESTED** [review]
Nice work! A few nits and suggestions.

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 2]
```diff
@@ -664,6 +664,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+file system can be registered with the catalog. Use the ``register_table``
+procedure on the catalog's ``system`` schema and supply the target schema,
+desired table name, and the location of the table metadata::
+
+    CALL iceberg.system.register_table('schema_name', 'table_name', 'hdfs://localhost:9000/path/to/iceberg/table/metadata/dir')
+
+.. note::
+
+    If multiple metadata files of the same version exist at the specified
+    location, the most recently modified one will be used.
```
[the remark this answers is not in the document]
Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst]: fixed!

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 3]
```diff
@@ -664,6 +664,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+
+A metadata file can optionally be included as an argument to ``register_table``
+in the case where a specific metadata file contains the targeted table state::
```
[the remark this answers is not in the document]
Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst]: fixed!

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 4]
```diff
@@ -664,6 +664,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+
+    CALL iceberg.system.register_table('schema_name', 'table_name', 'hdfs://localhost:9000/path/to/iceberg/table/metadata/dir', '00000-35a08aed-f4b0-4010-95d2-9d73ef4be01c.metadata.json')
+
+.. note::
+
+    When registering a table with the Hive metastore, the user calling the
+    procedure is set as the owner of the table and will have ``SELECT``,
```
[the remark this answers is not in the document]
Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst]: fixed

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 5]
```diff
@@ -664,6 +664,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+^^^^^^^^^^^^^^^^
+
+Iceberg tables can be unregistered from the catalog using the ``unregister_table``
+procedure on the catalog's ``system`` schema::
+
+    CALL iceberg.system.unregister_table('schema_name', 'table_name')
+
+.. note::
+
+    Table data and metadata will remain in the filesystem after a call to
```
[the remark this answers is not in the document]
Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst]: fixed

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 6]
```diff
@@ -664,6 +664,109 @@ example uses the earliest snapshot ID: ``2423571386296047175``
...
+    ``unregister_table`` only when using the Hive catalog. This is similar to
+    the behavior listed above for the ``DROP TABLE`` command.
```
[the remark this answers is not in the document]
Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst]: fixed

File: presto-docs/src/main/sphinx/connector/iceberg.rst [hunk 7]
```diff
@@ -915,91 +1018,6 @@ Drop the schema ``iceberg.web``::
...
-
-A metadata file can optionally be included as an argument to ``register_table``
-in the case where a specific metadata file contains the targeted table state::
-
-    CALL iceberg.system.register_table('schema_name', 'table_name', 'hdfs://localhost:9000/path/to/iceberg/table/metadata/dir', '00000-35a08aed-f4b0-4010-95d2-9d73ef4be01c.metadata.json')
-
-.. note::
-
-    The Iceberg REST catalog may not support table register depending on the
-    type of the backing catalog.
```
Comment (hantangwangd) [inline presto-docs/src/main/sphinx/connector/iceberg.rst:941]: I think we shouldn't simply discard the content added last time to resolve the conflict. This content is valid and should be added to the content you moved above.

Comment (jackychen718) [reply presto-docs/src/main/sphinx/connector/iceberg.rst:941]: added.

**Review by hantangwangd — APPROVED** [review]
Thanks for the fix. Change looks good to me. Please take this PR out of draft following @tdcmeehan 's guidance.

**Comment by tdcmeehan** [comment]
Looks good to me.  @steveburnett would you like to have a look?

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

Pull updated branch, reviewed new build of doc, all comments addressed and no new concerns found. Thanks!

**Review by tdcmeehan — APPROVED** [review]

**Review by ZacBlanco — APPROVED** [review]

**Pull request merged**